src/network/networkd-dhcp4.c

```diff
@@ -377,6 +377,23 @@ static int link_set_dhcp_routes(Link *link) {
                         return log_link_error_errno(link, r, "Could not set router: %m");
         }
 
+        Route *rt;
+        LIST_FOREACH(routes, rt, link->network->static_routes) {
+                if (!rt->gateway_from_dhcp)
+                        continue;
+
+                if (rt->family != AF_INET)
+                        continue;
```

Comment:
hmm, what about ipv6? if this is only implementable on ipv4, the man page should probably have a brief comment about that.